From f6b3b91899e62dd8dc6f121d9b2871158e4a2591 Mon Sep 17 00:00:00 2001
From: Owen Howard <owen@ziax.com>
Date: Tue, 4 Nov 2025 14:46:54 +0000
Subject: [PATCH] OKDify

---
 images/entrypoint.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/images/entrypoint.sh b/images/entrypoint.sh
index e1bd117..61881d4 100755
--- a/images/entrypoint.sh
+++ b/images/entrypoint.sh
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 # Always exit on errors.
-set -e
+set -eox pipefail
 
 # Give help text for parameters.
 usage()
@@ -58,6 +58,9 @@ get_source_folder_for_rhel_version()
         9)
         sourcedir=/usr/bin/rhel9
         ;;
+        10)
+        sourcedir=/usr/bin/rhel9
+        ;;
         *)
         echo "ERROR: RHEL Major Version Unsupported, rhelmajor=${rhelmajor}"
         ;;
-- 
2.51.1

