From d44d0f8f1addf4f4177032d6f2e33718e6167c18 Mon Sep 17 00:00:00 2001
From: Owen Howard <owen@ziax.com>
Date: Tue, 4 Nov 2025 14:46:54 +0000
Subject: [PATCH] OKDify

---
 images/entrypoint.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/images/entrypoint.sh b/images/entrypoint.sh
index 8c05698..de26a32 100755
--- a/images/entrypoint.sh
+++ b/images/entrypoint.sh
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 # Always exit on errors.
-set -ex
+set -eox pipefail
 
 # Give help text for parameters.
 usage()
-- 
2.54.0

