From 9d5bf09e6e237708e564c672c1153e2567c1ea95 Mon Sep 17 00:00:00 2001
From: Owen Howard <owen@ziax.com>
Date: Fri, 6 Jun 2025 12:44:25 +0100
Subject: [PATCH] OKDify

---
 images/entrypoint.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/images/entrypoint.sh b/images/entrypoint.sh
index 13f5496..6415890 100755
--- a/images/entrypoint.sh
+++ b/images/entrypoint.sh
@@ -36,7 +36,7 @@ get_source_folder_for_rhel_version()
                 rhelmajor="$(echo "$PLATFORM_ID" | sed -E 's/^platform:el([0-9]+)$/\1/')"
             fi
         ;;
-        rhel) rhelmajor=$(echo "${VERSION_ID}" | cut -f 1 -d .)
+        rhel|centos) rhelmajor=$(echo "${VERSION_ID}" | cut -f 1 -d .)
         ;;
         fedora)
             if [ "${VARIANT_ID}" = "coreos" ]; then
@@ -58,6 +58,9 @@ get_source_folder_for_rhel_version()
         9)
         sourcedir=/usr/bin/rhel9
         ;;
+        10)
+        sourcedir=/usr/bin/rhel9
+        ;;
         *)
         echo "ERROR: RHEL Major Version Unsupported, rhelmajor=${rhelmajor}"
         ;;
-- 
2.51.0

