From b116f166d3c4c02e1c19d5f4095580e3bbbc7d29 Mon Sep 17 00:00:00 2001
From: Owen Howard <owen@ziax.com>
Date: Fri, 12 Jun 2026 12:20:04 +0100
Subject: [PATCH] OKDify

---
 go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/go.mod b/go.mod
index 20cfaab46..389332f50 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/rook/rook
 
-go 1.24.10
+go 1.24.6
 
 replace (
 	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
-- 
2.54.0

