From 55089e996a08a53f5dcbdda59239ff5fcbda8e77 Mon Sep 17 00:00:00 2001
From: Owen Howard <owen@ziax.com>
Date: Wed, 4 Jun 2025 13:24:38 +0100
Subject: [PATCH] OKDify

---
 Containerfile.rollouts.plugin | 43 ++++++++++-------------------------
 1 file changed, 12 insertions(+), 31 deletions(-)

diff --git a/Containerfile.rollouts.plugin b/Containerfile.rollouts.plugin
index 6e9b5cb..15d8bef 100644
--- a/Containerfile.rollouts.plugin
+++ b/Containerfile.rollouts.plugin
@@ -18,13 +18,10 @@
 # Argo Rollouts Openshift Routes Plugin Build stage which performs the actual build of Argo Rollouts binaries
 ####################################################################################################
 
-FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_golang_1.22 AS rollouts-plugin-trafficrouter-openshift-build
+FROM registry.access.redhat.com/ubi9/go-toolset:1.22 AS rollouts-plugin-trafficrouter-openshift-build
 
-WORKDIR /go/src/github.com/argoproj/argo-rollouts
 
-COPY . .
-
-WORKDIR /go/src/github.com/argoproj/argo-rollouts/rollouts-plugin-trafficrouter-openshift/
+COPY --chown=default ./rollouts-plugin-trafficrouter-openshift .
 
 ENV GOFLAGS="-mod=mod"
 ENV CGO_ENABLED=1
@@ -37,13 +34,9 @@ RUN CGO_ENABLED=1 GOEXPERIMENT=strictfipsruntime go build -v -tags strictfipsrun
 # Argo Rollouts Build stage which performs the actual build of Argo Rollouts binaries
 ####################################################################################################
 
-FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_golang_1.22 AS argo-rollouts-build
-
-WORKDIR /go/src/github.com/argoproj/argo-rollouts
-
-COPY . . 
+FROM registry.access.redhat.com/ubi9/go-toolset:1.22 AS argo-rollouts-build
 
-WORKDIR /go/src/github.com/argoproj/argo-rollouts/argo-rollouts/
+COPY --chown=default ./argo-rollouts .
 
 ENV GOFLAGS="-mod=mod"
 ENV CGO_ENABLED=1
@@ -51,11 +44,11 @@ ENV GOEXPERIMENT=strictfipsruntime
 
 # Perform the build
 RUN CGO_ENABLED=1 GOEXPERIMENT=strictfipsruntime \
-go build -v -ldflags "-X github.com/argoproj/argo-rollouts/utils/version.version=${CI_ARGO_ROLLOUTS_UPSTREAM_TAG} \
+go build -v -ldflags "-X github.com/argoproj/argo-rollouts/utils/version.version=${ROLLOUTS_VERSION} \
   -X github.com/argoproj/argo-rollouts/utils/version.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ') \
-  -X github.com/argoproj/argo-rollouts/utils/version.gitCommit=${CI_ARGO_ROLLOUTS_UPSTREAM_COMMIT} \
+  -X github.com/argoproj/argo-rollouts/utils/version.gitCommit=${ROLLOUTS_COMMIT} \
   -X github.com/argoproj/argo-rollouts/utils/version.gitTreeState=clean \
-  -X github.com/argoproj/argo-rollouts.gitTag=${CI_ARGO_ROLLOUTS_UPSTREAM_TAG}" \
+  -X github.com/argoproj/argo-rollouts.gitTag=${ROLLOUTS_VERSION}" \
   -tags strictfipsruntime -o ./dist/rollouts-controller ./cmd/rollouts-controller
 
 ####################################################################################################
@@ -64,7 +57,7 @@ go build -v -ldflags "-X github.com/argoproj/argo-rollouts/utils/version.version
 ####################################################################################################
 # Argo Rollouts Base - used as the base for both the release and dev argo rollouts images
 ####################################################################################################
-FROM registry.access.redhat.com/ubi8/ubi-minimal AS argo-rollouts-base
+FROM quay.io/centos/centos:stream9-minimal AS argo-rollouts-base
 
 USER root
 RUN microdnf install shadow-utils  -y
@@ -97,26 +90,14 @@ USER 999
 WORKDIR /home/argo-rollouts
 
 # Create the binary for the default route plugin to the plugins directory
-COPY --from=rollouts-plugin-trafficrouter-openshift-build /go/src/github.com/argoproj/argo-rollouts/rollouts-plugin-trafficrouter-openshift/dist/rollouts-plugin-trafficrouter-openshift ./plugins/rollouts-trafficrouter-openshift/openshift-route-plugin
+COPY --from=rollouts-plugin-trafficrouter-openshift-build /opt/app-root/src/dist/rollouts-plugin-trafficrouter-openshift ./plugins/rollouts-trafficrouter-openshift/openshift-route-plugin
 
-COPY --from=argo-rollouts-build /go/src/github.com/argoproj/argo-rollouts/argo-rollouts/dist/rollouts-controller /usr/local/bin/
+COPY --from=argo-rollouts-build /opt/app-root/src/dist/rollouts-controller /usr/local/bin/
 
 LABEL \
-    name="openshift-gitops-1/argo-rollouts-rhel8" \
-    version=${CI_CONTAINER_VERSION} \
     License="Apache 2.0" \
-    com.redhat.component="openshift-gitops-argo-rollouts-container" \
-    com.redhat.delivery.appregistry="false" \
-    release=${CI_CONTAINER_RELEASE} \
-    upstream-version=${CI_UPSTREAM_VERSION} \
-    upstream-vcs-ref="${CI_ARGO_ROLLOUTS_UPSTREAM_COMMIT}" \
-    upstream-vcs-type="git" \
-    summary="Red Hat Openshift GitOps Argo Rollouts" \
-    io.openshift.expose-services="" \
-    io.openshift.tags="openshift,gitops,argorollouts" \
-    io.k8s.display-name="openshift-gitops-argo-rollouts" \
-    maintainer="William Tam <wtam@redhat.com>" \
-    description="Red Hat Openshift GitOps Argo Rollouts"
+    summary="OKD GitOps Argo Rollouts" \
+    description="OKD GitOps Argo Rollouts"
 
 ENTRYPOINT [ "/usr/local/bin/rollouts-controller"]
 
-- 
2.49.0

