From 3ec1cd7b030a644cfd2dae10a5d9bed28dc1a96c Mon Sep 17 00:00:00 2001
From: Owen Howard <owen@ziax.com>
Date: Wed, 4 Jun 2025 13:40:39 +0100
Subject: [PATCH] OKDify

---
 Containerfile.plugin | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/Containerfile.plugin b/Containerfile.plugin
index 0c4e957..dec6485 100644
--- a/Containerfile.plugin
+++ b/Containerfile.plugin
@@ -15,16 +15,18 @@
 # ------------------------------------------------------------------------
 
 # --- start build stage #1
-FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_golang_1.22 as builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.22 as builder
 
 COPY . /app
 WORKDIR /app/dex
 
+ARG VERSION=latest
+
 ENV GOFLAGS=""
-RUN GOOS=linux go build -mod=mod -o /tmp/dex -ldflags "-w -X main.version=$(git describe --tags --always --abbrev=7)" ./cmd/dex
+RUN GOOS=linux go build -mod=mod -o /tmp/dex -ldflags "-w -X main.version=${VERSION}" ./cmd/dex
 
 # --- start build stage #2
-FROM registry.access.redhat.com/ubi8/ubi-minimal
+FROM quay.io/centos/centos:stream9-minimal
 # Dex connectors, such as GitHub and Google logins require root certificates.
 # Proper installations should manage those certificates, but it's a bad user
 # experience when this doesn't work out of the box.
@@ -48,15 +50,8 @@ COPY --from=builder /app/dex/web /srv/dex/web
 USER 1001:1001
 
 LABEL \
-    name="openshift-gitops-1/dex-rhel8" \
     License="Apache 2.0" \
-    com.redhat.component="openshift-gitops-dex-container" \
-    com.redhat.delivery.appregistry="false" \
-    summary="Red Hat OpenShift GitOps Dex" \
-    downstream-vcs-type="git" \
-    upstream-vcs-type="git" \
-    io.openshift.expose-services="" \
-    io.openshift.tags="openshift,gitops,dex,sso" \
-    io.k8s.display-name="Red Hat OpenShift GitOps Dex" \
-    maintainer="William Tam <wtam@redhat.com>" \
-    description="Red Hat OpenShift GitOps Dex"
+    summary="OKD GitOps Dex" \
+    io.k8s.display-name="OKD GitOps Dex" \
+    maintainer="OKD Community <maintainers@okd.io>" \
+    description="OKD GitOps Dex"
-- 
2.49.0

